src/API/API.ts

```diff
@@ -0,0 +1,93 @@
+enum METHODS {
+  GET = 'GET',
+  POST = 'POST',
+  PUT = 'PUT',
+  DELETE = 'DELETE',
+}
+
+export type Headers = Record<string, string> & {
+  "set-cookie"?: string[]
+};
+
+type Options = {
+  method: METHODS;
+  data?: any;
+  timeout?: number
+  headers?: Headers
+};
+
+type OptionsWithoutMethod = Omit<Options, 'method'>;
+
+
+function queryStringify(data: Record<string, string>) {
+  if (typeof data !== 'object') {
+    throw new Error('Data must be object');
+  }
+
+  // Здесь достаточно и [object Object] для объекта
+  const keys = Object.keys(data);
+  return keys.reduce((result, key, index) => {
+    return `${result}${key}=${data[key]}${index < keys.length - 1 ? '&' : ''}`;
+  }, '?');
+}
```

Comment:
Мне кажется такая запись выглядела бы проще и лаконичнее
```suggestion
const queryStringify = (data) => {
  return '? ' + Object
    .entries(data)
    .map(([key, value]) => `${key}=${value}`)
    .join('&')
}
```

Reply:
done